libs/core/algorithms/include/hpx/parallel/util/loop.hpp

```diff
@@ -104,11 +105,12 @@ namespace hpx::parallel::util {
 
     ///////////////////////////////////////////////////////////////////////////
     namespace detail {
-
+        template <typename Iterator, typename ExPolicy>
+        struct loop_pred;
         // Helper class to repeatedly call a function starting from a given
         // iterator position till the predicate returns true.
         template <typename Iterator>
-        struct loop_pred
+        struct loop_pred<Iterator, void>
```

Comment:
Where is this used? Is `std::is_same_v<Expolicy, void>` true at any point?

Reply:
I believe this was done so loop_pred can be templated as `loop<Iter>` and `loop<Iter, ExPolicy>`
Now that we have moved the dispatch to tag_invoke, I don't think this is required, I will look into replacing it with the older version.